tenants-attributes-checker/src/main/scala/it/pagopa/interop/tenantsattributeschecker/util/ReadModelQueries.scala

```diff
@@ -0,0 +1,46 @@
+package it.pagopa.interop.tenantsattributeschecker.util
+
+import it.pagopa.interop.commons.cqrs.service.MongoDbReadModelService
+import it.pagopa.interop.tenantmanagement.model.persistence.JsonFormats.ptFormat
+import it.pagopa.interop.tenantmanagement.model.tenant.PersistentTenant
+import it.pagopa.interop.tenantsattributeschecker.ApplicationConfiguration._
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates._
+import org.mongodb.scala.model.Filters
+import org.mongodb.scala.model.Filters._
+
+import scala.concurrent.Future
+
+object ReadModelQueries {
+
+  def getAllExpiredAttributesTenants(
+    rm: MongoDbReadModelService,
+    expirationInMonth: Boolean = false
+  ): Future[Seq[PersistentTenant]] =
+    getAll(100)(getExpiredAttributesTenants(_, _, rm, expirationInMonth))
+
+  private def getExpiredAttributesTenants(
+    offset: Int,
+    limit: Int,
+    readModelService: MongoDbReadModelService,
+    expirationInMonth: Boolean
+  ): Future[Seq[PersistentTenant]] = {
+
+    val dateFilter: Bson =
+      if (expirationInMonth)
+        Filters.eq("data.attributes.verifiedBy.extensionDate", dateTimeSupplier.get().plusDays(30).toString)
```

Comment:
In this case we'll get all the attributes that will expire in a month (I know that the variable was confusing), shouldn't we get only the ones who expire exactly after 30 days from now?

Otherwise we would send a notification every day for a month for those attributes.

Reply:
you're right, we should avoid to send notifications every time the job runs.
I see 2 edge cases using the equality:
1. the job does not run for 1+ days -> notifications are not sent
2. the job runs multiple times a day -> multiple notifications are sent

@beetlecrunch what do you think? are the two cases acceptable? 
Unfortunately the only alternative that I can think of requires to persist if each notification has been sent or not, that I would avoid if possible